x/txfees/keeper/hooks.go

```diff
@@ -0,0 +1,64 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	epochstypes "github.com/osmosis-labs/osmosis/v7/x/epochs/types"
+	txfeestypes "github.com/osmosis-labs/osmosis/v7/x/txfees/types"
+)
+
+func (k Keeper) BeforeEpochStart(ctx sdk.Context, epochIdentifier string, epochNumber int64) {}
+
+// at the end of each epoch, swap all non-OSMO fees into OSMO and transfer to fee module account
+func (k Keeper) AfterEpochEnd(ctx sdk.Context, epochIdentifier string, epochNumber int64) {
+	addrNonNativeFee := k.accountKeeper.GetModuleAddress(txfeestypes.NonNativeFeeCollectorName)
+	nonNativeFeeAccountBalances := k.bankKeeper.GetAllBalances(ctx, addrNonNativeFee)
+	baseDenom, _ := k.GetBaseDenom(ctx)
+
+	for _, coin := range nonNativeFeeAccountBalances {
+		if coin.Denom == baseDenom {
+			continue
+		} else {
+			feetoken, err := k.GetFeeToken(ctx, coin.Denom)
+			if err != nil {
+				panic(err)
+			}
+
+			// We allow full slippage. Theres not really an effective way to bound slippage until TWAP's land,
+			// but even then the point is a bit moot.
+			// The only thing that could be done is a costly griefing attack to reduce the amount of osmo given as tx fees.
+			// However the idea of the txfees FeeToken gating is that the pool is sufficiently liquid for that base token.
+			minTokenOut := sdk.ZeroInt()
+			_, err = k.gammKeeper.SwapExactAmountIn(ctx, addrNonNativeFee, feetoken.PoolID, coin, baseDenom, minTokenOut)
```

Comment:
There's no need to set a `minTokenOut` variable as it's only used once (below).
```suggestion
			_, err = k.gammKeeper.SwapExactAmountIn(ctx, addrNonNativeFee, feetoken.PoolID, coin, baseDenom, sdk.ZeroInt())
```